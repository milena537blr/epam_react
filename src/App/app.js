import React from "react";
// import PropTypes from 'prop-types';
import Article from "../Article/Article";
import Search from "../Search/Search";
import data from "../data/data.json";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
import s from "./app.module.scss";
import Logo from "../Logo/Logo";
import Box from "../Box/Box";
import Button from "../Button/Button";
import Card from "../Card/Card";

class App extends React.Component {
  constructor(props) {
    super(props);

    let cards = data.cards.map(item => {
      return {
        name: item.name,
        date: item.date,
        genre: item.genre,
        id: item.id
      };
    });

    this.state = {
      data,
      listCards: [],
      searchText: "",
      cards,
      isSearchActive: true,
      currentCard: {},
      filterText: ""
    };
  }

  handleSearchTextChange = searchText => {
    this.setState({ searchText });
  };

  handleCardClick = currentCardId => {
    this.setState({
      isSearchActive: false
    });

    this.state.data.cards.forEach(currentCard => {
      if (currentCard.id === currentCardId) {
        this.setState({ currentCard });
        return;
      }
    });
  };

  handleSearchClick = () => {
    this.setState({
      isSearchActive: true
    });
  };

  render() {
    const isSearchActive = this.state.isSearchActive;

    const cards = this.state.cards.map(card => {
      return (
        <Card
          cardClick={this.handleCardClick}
          card={card}
          key={card.id}
        />
      );
    });

    return (
      <ErrorBoundary>
        <header className={s.header}>
          <div className={s.overlay} />
          <section className={s.container}>
            <Box align="space-between" verticalAlign="middle" marginBottom={8}>
              <Logo />
              <Button
                onHandleSearchClick={this.handleSearchClick}
                text="SEARCH"
                size="large"
                color="white"
              />
            </Box>
            {isSearchActive ? (
              <Search
                searchText={this.state.searchText}
                onSearchTextChange={this.handleSearchTextChange}
              />
            ) : (
              <Article card={this.state.currentCard} />
            )}
          </section>
        </header>
        <section className={s.panel}>
          <Box
            align="space-between"
            verticalAlign="middle"
            className={s.container}
          >
            <div>7 movies found</div>
            <div>
              <span>Sort by </span>
              <span>release date </span>
              <span>rating</span>
            </div>
          </Box>
        </section>
        <main className={s.main}>
          <div className={s.container}>
            <Box className={s.wrapper} align="center">
              {cards}
            </Box>
          </div>
        </main>
        <footer className={s.footer}>
          <div className={s.container}>
            <Logo />
          </div>
        </footer>
      </ErrorBoundary>
    );
  }
}

/* App.propTypes = {
  name: PropTypes.string,
}; */

export default App;
