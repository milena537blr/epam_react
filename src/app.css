.app {
  margin: 1rem;
  font-family: Arial, Helvetica, sans-serif;
}

h1, h2, h3, h4, h5, p {
  font-family: helvetica;
  color: #3e3e3e;
}
.description {
  font-size: 14px;
  color: #9e9e9e;
}
.awful-selfie{
  background: url('cat.jpg');
  width: 300px;
  height: 300px;
  background-size: 100% auto;
  background-repeat: no-repeat;
}