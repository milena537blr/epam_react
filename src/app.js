import React from "react";
import PropTypes from "prop-types";
import { Header } from "./components/Header/Header";

class Panel extends React.Component {
  render() { 
    return (
      <div>{this.props.name}</div>
    );
  }
}

class Main extends React.Component {
  render() { 
    return (
      <div>{this.props.name}</div>
    );
  }
}

class Footer extends React.Component {
  render() { 
    return (
      <div>{this.props.name}</div>
    );
  }
}

Panel.propTypes = {
  name: PropTypes.string
};

Main.propTypes = {
  name: PropTypes.string
};

Footer.propTypes = {
  name: PropTypes.string
};

class App extends React.Component {
  render() {
    return (
      <div>
        <Header name="netflixroulette" />
        <Panel name="panel" />
        <Main name="main" />
        <Footer name="footer" />
      </div>
    );
  }
}

export { App };
