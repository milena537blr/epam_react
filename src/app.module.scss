@import "./styles/core";

.header {
  padding: rem(20) rem(40);
  background-image: url("./images/header-bg.jpg");
  min-height: 400px;
  position: relative;

  .overlay {
    background-color: rgba(0, 0, 0, 0.7);
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }

  &__searchTitle {
    margin-bottom: rem(20);
    margin-top: rem(40);
  }
}

.container {
  @include container;
}

.footer {
  background-color: darken($color--gray, 35%);
  padding: rem(20) 0;
}

.container {
  @include container;
}

.topBar {
  display: flex;
  justify-content: space-between;
  margin-bottom: $grid * 8;
}

.panel {
  background-color: lighten($color--gray, 40%);
  padding: rem(10);
}

.main {
  background-color: $color--white;
}

.wrapper {
  padding: rem(40) rem(20);
  display: flex;
  flex-wrap: wrap;
}