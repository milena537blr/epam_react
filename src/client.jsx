import React from "react";
import { hydrate } from "react-dom";
import { BrowserRouter } from "react-router-dom";
import Root from "./Root";
import { configureStore } from "../src/store/configureStore";
import { configurePersistor } from "../src/store/configureStore";
import { createBrowserHistory } from "history";

const store = configureStore(window.PRELOADED_STATE);
const persistor = configurePersistor(window.PRELOADED_STATE);
const history = createBrowserHistory();

const root = <Root Router={BrowserRouter} store={store} persistor={persistor} history={history} />;

hydrate(root, document.getElementById("root"));
