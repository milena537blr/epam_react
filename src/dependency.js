export function someFunction(){
  console.log(13);
}