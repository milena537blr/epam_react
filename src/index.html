<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <title>React App</title>
</head>

<body>
  <h1>Expack111</h1>
  <p class="description">Lorem ipsum</p>
  <div class="awful-selfie"></div>
  <div id="root1"></div>
  <div id="root2"></div>
  <noscript>
    You need to enable JavaScript to run this app.
  </noscript>
</body>

</html>