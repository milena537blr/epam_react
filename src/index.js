import React from "react";
import ReactDOM from "react-dom";
import { App, Component3 } from "./app.js";

ReactDOM.render(
  <App />,
  window.document.getElementById('root1')
)

ReactDOM.render(
  React.createElement(Component3, { name: 'React.createElement' }, null),
  window.document.getElementById('root2')
);

// eslint-disable-line no-undef 
