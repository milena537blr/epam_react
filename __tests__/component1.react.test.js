'use strict';

import React from 'react';
import Component1 from '../src/app';
import renderer from 'react-test-renderer';
import "../src/app";

it('renders correctly', () => {
  const component = renderer
    .create(<Component1 name="functional component" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});