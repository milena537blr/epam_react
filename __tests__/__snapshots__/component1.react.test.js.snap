exports[`Component1 renders correctly`] = `
<h1>
  Hello functional component!
</h1>
`;